### DISCUSSION TITLE
POST request body of empty list is mutated to empty dict

### DISCUSSION THREAD

**Discussion by topleft** [fastapi/fastapi#6253] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Dict

from fastapi import Body, FastAPI, Query
from fastapi.testclient import TestClient

app = FastAPI()


@app.post("/query")
def read_query(q: Dict):
    return q


client = TestClient(app)

def test_empty_list_invalid():
    response = client.post("/query", json=[])
    assert response.status_code == 422

def test_empty_list():
    response = client.post("/query", json=[])
    assert response.json() == []

def test_empty_list_does_not_mutate_to_dict():
    response = client.post("/query", json=[])
    assert response.json() != {}
```

### Description

**What**
Defining a POST request endpoint and setting the type for the Body to a `dict` type, then calling that endpoint with with an empty list for the body.

**Expectation**
I expect that calling the endpoint with an empty list results in  `response.status_code = 422`.

**Current Behavior**
What is currently happening is that the empty list is mutated into an empty dict.

### Operating System

macOS

### FastAPI Version

0.78.0

### Python Version

Python 3.10.4

### Additional Context

I opened a PR in my fork with the above code snippet: https://github.com/topleft/fastapi/pull/1/files.

Thanks for creating and maintaining FastAPI!

**Comment by JarroVGIT** [comment, marked as answer]
I think this is in the nature of Python and how it casts to certain types. FastAPI uses this to validate any incoming request. In your case, FastAPI will try to cast `q` as `dict(q)`. When providing an empty list to `dict()`, it will return an empty dictionary `{}`. 
```
>>> dict([])
{}
```
There is little FastAPI can do about this. Even if you were to put it into a Pydantic model, empty lists would still be casted to an empty dict:
```python
from fastapi import FastAPI
from pydantic import BaseModel

app = FastAPI()

class SomeModel(BaseModel):
    q: dict

@app.post("/query")
def read_query(q: SomeModel):
    return q
```
Curl:
```
% curl -X POST localhost:8000/query -d '{"q":[]}' -H 'Content-Type: application/json'
{"q":{}}%      
```

**Comment by topleft** [comment]
Thanks @JarroVGIT. That makes sense to me. Appreciate you taking a look.
